src/main/java/org/threeten/extra/chrono/InternationalFixedDate.java

```diff
@@ -0,0 +1,887 @@
+    @Override  // for covariant return type
+    @SuppressWarnings ("unchecked")
+    public ChronoLocalDateTime<InternationalFixedDate> atTime(final LocalTime localTime) {
+        return (ChronoLocalDateTime<InternationalFixedDate>) ChronoLocalDate.super.atTime(localTime);
+    }
+
+    /**
+     * {@inheritDoc}
+     * @return number of TemporalUnit until given Temporal
+     */
+    @Override
+    public long until(final Temporal endExclusive, final TemporalUnit unit) {
+        InternationalFixedDate end = InternationalFixedDate.from(endExclusive);
+
+        if (unit instanceof ChronoUnit) {
+            switch ((ChronoUnit) unit) {
+                case DAYS:
+                    return daysUntil(end);
+                case WEEKS:
+                    return daysUntil(end) / InternationalFixedChronology.DAYS_IN_WEEK;
+                case MONTHS:
+                    return monthsUntil(end);
+                case YEARS:
+                    return monthsUntil(end) / InternationalFixedChronology.MONTHS_IN_YEAR;
+                case DECADES:
+                    return monthsUntil(end) / (InternationalFixedChronology.MONTHS_IN_YEAR * 10);
+                case CENTURIES:
+                    return monthsUntil(end) / (InternationalFixedChronology.MONTHS_IN_YEAR * 100);
+                case MILLENNIA:
+                    return monthsUntil(end) / (InternationalFixedChronology.MONTHS_IN_YEAR * 1000);
+                case ERAS:
+                    return end.getLong(ChronoField.ERA) - getLong(ChronoField.ERA);
+            }
+
+            throw new UnsupportedTemporalTypeException("Unsupported unit: " + unit);
+        }
+
+        return unit.between(this, end);
+    }
+
+    /**
+     * Get the number of years from this date to the given day.
+     *
+     * @param end The end date.
+     * @return The number of years from this date to the given day.
+     */
+    long yearsUntil(final InternationalFixedDate end) {
+        long startYear = getProlepticYear() * 512L + getDayOfYear();
+        long endYear = end.getProlepticYear() * 512L + end.getDayOfYear();
+
+        return (endYear - startYear) / 512L;
+    }
+
+    /**
+     * {@inheritDoc}
+     * @param endDateExclusive
+     * @return period up to that date
+     */
+    @Override
+    public ChronoPeriod until(final ChronoLocalDate endDateExclusive) {
+        InternationalFixedDate end = InternationalFixedDate.from(endDateExclusive);
+        int years = Math.toIntExact(yearsUntil(end));
+        // Get to the same "whole" year.
+        InternationalFixedDate sameYearEnd = end.plusYears(years);
+        int months = (int) monthsUntil(sameYearEnd);
+        int days = (int) daysUntil(sameYearEnd.plusMonths(months));
+
+        return getChronology().period(years, months, days);
+    }
+
+    /**
+     * {@inheritDoc}
+     * @param end
+     * @return period up to that date
+     */
+    @Override
+    long monthsUntil(final AbstractDate end) {
```

Comment:
I get the feeling this is off somehow, although I haven't run enough of the numbers to find where.  In particular, you can't blanket translate leap-day/year-day like you do for day-of-month here, it's more complicated than that.  See Discordian, again.